tests/integration/test_exodus.py

```diff
@@ -0,0 +1,81 @@
+import requests
+
+
+def test_exodus_basic(cdn_test_url):
+    url = (
+        cdn_test_url
+        + "/content/beta/rhel8/8/x86_64/exodus_test/os/bash-4.4.19-7.el8.x86_64.rpm"
+    )
+
+    r = requests.get(url)
+    assert r.status_code == 200
+    assert "cache-control" not in r.headers
+
+
+def test_header_not_exist_file(cdn_test_url):
+    url = (
+        cdn_test_url
+        + "/content/beta/rhel8/8/x86_64/exodus_test/os/bash-4.4.19-7.el8.x86_64.rpm_no"
+    )
+    r = requests.get(url)
+    assert r.status_code == 404
+    assert "cache-control" not in r.headers
+
+
+def test_header_cache_control(cdn_test_url):
+    patterns = [
+        "/content/beta/rhel8/8/x86_64/exodus_test/os/listing",
+        "/content/beta/rhel8/8/x86_64/exodus_test_ctype/os/repodata/repomd.xml",
+        "/content/beta/rhel8/8/x86_64/exodus_test/os/ostree/repo/refs/heads/sub_dir/testing",
+    ]
+    multiple_url = [cdn_test_url + pattern_item for pattern_item in patterns]
+    for url in multiple_url:
+        r = requests.get(url)
+        assert r.status_code == 200
+        assert r.headers["cache-control"] == "max-age=600"
+
+
+def test_header_want_digest_get(cdn_test_url):
+    headers = {"want-digest": "sha256"}
+    url = cdn_test_url + "/content/beta/rhel8/8/x86_64/exodus_test/os/listing"
+    r = requests.get(url, headers=headers)
+    assert r.status_code == 200
+    assert "digest" in r.headers
+
+
+def test_header_want_digest_head(cdn_test_url):
+    headers = {"want-digest": "sha256"}
+    url = cdn_test_url + "/content/beta/rhel8/8/x86_64/exodus_test/os/listing"
+    r = requests.head(url, headers=headers)
+    assert r.status_code == 200
+    assert "digest" in r.headers
```

Comment:
I think we should verify the digests in these tests since our code is responsible for creating the header.

Reply:
I agree.